src/zarr/core/metadata/v2.py

```diff
@@ -31,7 +32,7 @@ class ArrayV2Metadata(ArrayMetadata):
     shape: ChunkCoords
     chunk_grid: RegularChunkGrid
     data_type: np.dtype[Any]
-    fill_value: None | int | float = 0
+    fill_value: None | int | float | str | bytes = 0
```

Comment:
Need to confirm the expectation here: is `ArrayV2Metadata` supposed to be the in-memory version of `fill_value` (e.g. `b"X"`) or the on-disk version (e.g. `"WA=="`). Check other dtypes.

Reply:
Based on the handling of `complex`-type `fill_value`, I think this is supposed to be the "in-memory" version rather than the on-disk version. So I think `bytes` should be allowed here.